src/Groups/matrices/linear_centralizer.jl

```diff
@@ -0,0 +1,395 @@
+########################################################################
+#
+# Orders
+#
+########################################################################
+
+function _GL_order(n::Int, q::fmpz)
+   res = q^div(n*(n-1),2)
+   for i in 1:n res *= (q^i-1) end
+   return res
+end
+
+_GL_order(n::Int, F::Ring) = _GL_order(n, order(F))
+
+function _SL_order(n::Int, q::fmpz)
+   res = q^div(n*(n-1),2)
+   for i in 2:n res *= (q^i-1) end
+   return res
+end
+
+_SL_order(n::Int, F::Ring) = _SL_order(n, order(F))
+
+########################################################################
+#
+# Centralizer in GL
+#
+########################################################################
+
+
+# returns as matrices
+function _gens_for_GL(n::Int, F::Ring)
+   n !=1 || return [matrix(F,1,1,[primitive_element(F)])]
+   if order(F)==2
+      h1 = identity_matrix(F,n)
+      h1[1,2] = 1
+      h2 = zero_matrix(F,n,n)
+      h2[1,n] = 1
+      for i in 1:n-1 h2[i+1,i] = -1 end
+      return h1,h2
+   else
+      h1 = identity_matrix(F,n)
+      h1[1,1] = primitive_element(F)
+      h2 = zero_matrix(F,n,n)
+      h2[1,1] = -1
+      h2[1,n] = 1
+      for i in 1:n-1 h2[i+1,i] = -1 end
+      return h1,h2
+   end
+end
```

Comment:
These are the generators from [the paper by Taylor](https://www.gap-system.org/Manuals/doc/ref/chapBib.html#biBTay87), which are used also in GAP. In would propose to give a reference to this paper.